src/ciphers/caesar.rs

```diff
@@ -1,20 +1,25 @@
-//! Caesar Cipher
-//! Based on cipher_crypt::caesar
-//!
-//! # Algorithm
-//!
-//! Rotate each ascii character by shift. The most basic example is ROT 13, which rotates 'a' to
-//! 'n'. This implementation does not rotate unicode characters.
-
-/// Caesar cipher to rotate cipher text by shift and return an owned String.
-pub fn caesar(cipher: &str, shift: u8) -> String {
-    cipher
-        .chars()
+/// Encrypts a given text using the Caesar cipher technique.
+///
+/// In cryptography, a Caesar cipher, also known as Caesar's cipher, the shift cipher, Caesar's code,
+/// or Caesar shift, is one of the simplest and most widely known encryption techniques.
+/// It is a type of substitution cipher in which each letter in the plaintext is replaced by a letter
+/// some fixed number of positions down the alphabet.
+///
+/// # Arguments
+///
+/// * `text` - The text to be encrypted.
+/// * `rotation` - The number of rotations (shift) to be applied.
+///
+/// # Returns
+///
+/// Returns the encrypted string.
+pub fn caesar(text: &str, rotation: usize) -> String {
+    text.chars()
         .map(|c| {
             if c.is_ascii_alphabetic() {
                 let first = if c.is_ascii_lowercase() { b'a' } else { b'A' };
-                // modulo the distance to keep character range
-                (first + (c as u8 + shift - first) % 26) as char
+
+                (((c as u8 - first) + (rotation % 26) as u8) % 26 + first) as char
             } else {
                 c
```

Comment:
Since this is the essence of the entire cipher, I would move it into a separate function like `rotate_char` or `shift_char`. What do you think?

Reply:
Great idea, this will make the function much more readable. Consider it done.